### PR TITLE
Wallet fiat total don't get refreshed after enabling/disabling chain until app is restarted #3738

### LINKED ISSUE #3738: Wallet fiat total don't get refreshed after enabling/disabling chain until app is restarted
If we have $E amount of crypto in chain Ethereum mainet and $G in Gnosis:

1. Make sure app only has Ethereum mainnet and Gnosis enabled
2. Kill app
3. Launch app
4. Verify fiat total shown in Wallet tab header is E + G
5. Disable Ethereum mainnet so only Gnosis is active

Expect:

6. Total = G

Observed:

6. Total = E + G (until restart app)

eg. "Fiat total" in screenshot is "$2.63":

Screenshot 2022-01-18 at 12 32 54 AM

### PR DESCRIPTION
**Pull request by oa-s**
Closes #3738

### CODE REVIEW

File: AlphaWallet/Core/Coordinators/WalletBalance/WalletBalanceFetcher.swift [hunk 1]
```diff
@@ -122,6 +122,13 @@ class WalletBalanceFetcher: NSObject, WalletBalanceFetcherType {
         for each in deletedServers {
             services.remove(at: each)
         }
+
+        queue.async { [weak self] in
```
Comment (hboon) [inline AlphaWallet/Core/Coordinators/WalletBalance/WalletBalanceFetcher.swift:126]: Is the `queue.async` necessary?

Comment (oa-s) [reply AlphaWallet/Core/Coordinators/WalletBalance/WalletBalanceFetcher.swift:126]: added it as the rest of calls of this method called async.

**Comment by oa-s** [comment]
> There is a lot of jumping from the custom dispatch queue back to the main queue and back again. It's very hard to follow. Is there a diagram or something to help understand the code?

nope, we don't have such kind of diagram, i will recheck this issue later

**Review by hboon — APPROVED** [review]
